impeller/docs/faq.md

```diff
@@ -100,3 +100,113 @@
     as development progresses.
 * How do you run `impeller_unittests` with Playgrounds enabled?
   * Specify the `--enable_playground` command-line option.
+* Describe Impeller in a Tweet.
+  * "Impeller is AOT mode for rendering."
+* Why did the Flutter team build Impeller?
+  * The short answer, for consistent performance. The longer answer...
+  * Flutter applications, via their use of Skia, were susceptible to the problem
+    of jank due to shader compilation. Such problems were [reported as early as
+    2015](https://github.com/flutter/flutter/issues/813).
+  * Shaders are tiny programs that run on the GPU when a specific combination of
+    rendering intent is requested by the application. Flutters API is extremely
+    expressive and it was thought to be impossible to determine the entire set
+    of shaders an application might need at runtime. So, applications would
+    create and compile these shaders just-in-time as they were needed. Once they
+    were compiled, they could be cached.
+  * But this led to the problem of jank during the initial run of the
+    application when there were no cached shaders.
+  * Fortunately, early on, Flutter applications were rather simple. And the team
+    was able to predict the common shaders the application was likely going to
+    need. A generic ["warm up"](https://github.com/flutter/flutter/pull/27660)
+    phase was added where these shaders were given a chance to be compiled and
+    cached at the launch of a Flutter application.
+  * As Flutter applications began to become more complex, the generic shader
+    warmup was no longer suitable for all apps. So the ability to specify
+    shaders to warm up was [exposed to end
+    users](https://api.flutter.dev/flutter/painting/ShaderWarmUp-class.html).
+  * But, this was exceedingly hard to do and also platform specific (iOS Metal,
+    Android OpenGL, Fuchsia Vulkan, etc.). So much so that only a handful of
+    engineers (who all worked on the Flutter Engine) could effectively build the
+    rendering intent that would warm up the right shaders. And these shaders
+    were specific to the app version. If the app was updated to add or remove
+    more features, the warmup routines were invalid and mostly just served to
+    slow down application startup.
+  * To make this process self-serve, [a scheme was
+    devised](https://github.com/flutter/flutter/issues/53607) to run the
+    application and its tests during development in training phase, capture the
+    shaders, package them with the application, and during a new "warm up"
+    phase,  pre-compile the shaders.
+  * But, remember these shaders were platform specific? This moved the burden of
```

Comment:
remove the ?

Just: Remember, these shaders were platform specific. The scheme detailed above unfortunately moved the burden to developers.

Reply:
Done.